### DISCUSSION TITLE
Use runtime config in nuxt.config.ts

### DISCUSSION THREAD

**Discussion by vionijannet** [nuxt/nuxt#21075] [Questions]
Hi everyone, is there anyone know how to use runtime config in nuxt config itself?
So the background is I use library @nuxt-alt/proxy as proxy so the network will not expose API service
I put API in the runtime config, and the proxy settings is also in nuxt.config
The proxy need to know the API, but I don't know how to call runtime config from itself?

So here's my nuxt config, and also runtime config
```
runtimeConfig: {
    public: {
        api: "https://the-service.com"
    }
},
proxy: {
    proxies: {
        '/proxy/api': {
            target: 'https://the-service.com', // this should call runtime config
            changeOrigin: true,
            secure: false,
        },
    }
},
```

I tried to change the target to `target: useRuntimeConfig().public.api`, but it made error after restart nuxt. 
The error said `[unhandledRejection] Nuxt instance is unavailable!`.

I really appreciate for the response, thankyou ☺

**Comment by maou-shonen** [comment]
You can use in development environment
```ts
target: process.env.API_URL || 'https://example.com'
```
Currently in production environment,
Unless the module supports it or you can override it with a plugin,
otherwise it seems impossible.
[see explanation here](https://github.com/unjs/nitro/issues/1161#issuecomment-1511444675)

**Reply by vionijannet** [reply]
Do I need to create file .env?

**Comment by martinszeltins** [comment]
> You can use in development environment
> 
> ```ts
> target: process.env.API_URL || 'https://example.com'
> ```
> 
> Currently in production environment, Unless the module supports it or you can override it with a plugin, otherwise it seems impossible. [see explanation here](https://github.com/unjs/nitro/issues/1161#issuecomment-1511444675)

This makes it impossible to override it using environment configuration at runtime because the app is built with API_URL as it was at build-time. If you wanted to change API_URL on some environment, you would have to rebuild the whole app. After the app is built, you cannot change the API_URL environment anymore.
